### ISSUE TITLE
Option to create approved file if non exists by ShouldMatchApproved()

### ISSUE THREAD

**Issue by Piedone** [shouldly/shouldly#435]
Currently if ShouldMatchApproved() is called but no approved file exists one is presented with a difftool window. It would be great to be able to choose that Shouldly instead saves the first result (after any scrubbing) as the approved file without starting a difftool (but the test would still fail first, not to give a false positive). So the result will be the same that happens when you run the command Shouldly tells you to run after it can't find an approval file.

Our use-case behind this is as following: our project (which BTW is [Hastlayer](https://hastlayer.com) that converts .NET software into computer chips, check it out if you do scientific computing) generates source files as part of its outputs and we verify these. Now if some change is made to the app logic that invalidates dozens of approval files it would be much easier to just remove them, regenerate them in one go and check them at one glance before committing the changes. Often code generation logic changes affect a lot of approval files but the diffs are trivial (like some wording in a generated comment changing) so it's a hassle to re-save them one by one.

If you can agree then I'd venture into submitting a PR for this. (I think only ShouldMatchApproved(), ShouldMatchConfigurationBuilder and ShouldMatchConfiguration needs to be changed for this, and a test case added to ShouldMatchApprovedScenarios.)

And BTW thank you for this great library! I find joy in writing test code again!

**Comment by jnm2** [comment]
I like this idea. If you run the test suite twice, the tests will first fail then succeed, so that could be weird. But source control should show you all the added files. Maybe we could have an opt-in setting to turn on this behavior.

@SimonCropp You know more about the diffing engine than I do. Is this already supported? And would you have an alternative suggestion that would be more idiomatic while still improving their scenario?

**Comment by Piedone** [comment]
I'd definitely do this optionally, in an opt-in manner, only, not by default.

**Comment by jnm2** [comment]
A question that remains for me is whether it's idiomatic in snapshot testing to simply turn on a setting temporarily and run the tests to regenerate all files rather than deleting all or selectively deleting, then rerunning.

**Comment by SimonCropp** [comment]
no it is not currently available. 

i have found with diffenginetray https://github.com/VerifyTests/DiffEngine/blob/master/docs/tray.md it is trivial to approve new files when they popup

**Comment by jnm2** [comment]
Okay, so we'll have to file an issue at https://github.com/VerifyTests/DiffEngine and possibly contribute a fix if we want to implement this new option.

@Piedone @SimonCropp What do both of you think of exposing a different setting, one that overwrites all existing files which you only turn on temporarily, run all tests, and turn back off? `UpdateMode = true` or something like that?

**Comment by Piedone** [comment]
I always feel uneasy about adding temporary changes to code that is otherwise kept in a repository (so not like an untracked config file). Being able to recreate removed verification files seems like a more natural way of handling this.

**Comment by SimonCropp** [comment]
> Okay, so we'll have to file an issue at https://github.com/VerifyTests/DiffEngine and possibly contribute a fix if we want to implement this new option.

I dont think this is the correct approach. the api for diffengine is essentially "show this diff", if the caller wants that diff to be accepted, then there is no diff to show. I think we should focus on how make accepting diffs easier.

 In Verify the default behavior is to [place all "accept diffs commands" into the clipboard](https://github.com/VerifyTests/Verify/blob/master/docs/clipboard.md). then if the user thinks they all look good, they can then be pasted into a command windows to accept them all. ApprovalTests has similar functionality but the clipboard behavior is opt in.  [DiffEngineTray](https://github.com/VerifyTests/DiffEngine/blob/master/docs/tray.md) is an alternative to the clipboard approach and if DiffEngineTray is running the clipboard approach is disabled. DiffEngineTray also has keyboard shortcuts for accepting diffs

AFAIK shouldly lacks any of these higher level accept mechanism, so it is friction to approve 1 file and that friction increases with more files.  

What do you think about adopting a similar approach to Verify or ApprovalTests?
